File: CPP/Problem006.cpp
#include "iostream"
#include "math.h"

int Problem006();

int main() {
    clock_t start; 
    start = std::clock();
    int ans = Problem006();
    double time_taken = double(clock() - start) / double(CLOCKS_PER_SEC); 

    printf("\nAnswer: %d\n", ans);
    printf("Time Taken: %f seconds\n\n", time_taken);

    return 0;
}

/*
Problem006 answers the problem at : https://projecteuler.net/problem=6

* Problem 6:
		Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.
*/
int Problem006(){
    int sqSum = 0, sumSq = 0;

    for (int i = 1; i <= 100; i++){
        sqSum += pow(i, 2);
        sumSq += i;
    }

    sumSq = pow(sumSq, 2);    
    return abs(sqSum - sumSq);
}

File: Python/Problem006.py
from time import time

''' 
Problem 6: https://projecteuler.net/problem=6

Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.
'''

start = time()
sqSum = lambda n: int(n * (n+1) * (2*n + 1) / 6)
sumSq = lambda n: int(n * (n+1) / 2) ** 2

ans = sumSq(100) - sqSum(100)

if __name__ == '__main__':
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")